edge-util/src/Microsoft.Azure.Devices.Edge.Util/CertificateHelper.cs

```diff
@@ -207,13 +208,17 @@ public static bool ValidateCertExpiry(X509Certificate2 certificate, ILogger logg
             return true;
         }
 
-        public static void InstallCerts(StoreName name, StoreLocation location, IEnumerable<X509Certificate2> certs)
+        public static void InstallCertificates(IEnumerable<X509Certificate2> certificateChain, ILogger logger)
         {
-            List<X509Certificate2> certsList = Preconditions.CheckNotNull(certs, nameof(certs)).ToList();
-            using (var store = new X509Store(name, location))
+            X509Certificate2[] certs = Preconditions.CheckNotNull(certificateChain, nameof(certificateChain)).ToArray();
+
+            StoreName storeName = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? StoreName.CertificateAuthority : StoreName.Root;
+
+            logger?.LogInformation($"Installing certificates {string.Join(",", certs.Select(c => $"[{c.Subject}:{c.GetExpirationDateString()}]"))} to {storeName}");
```

Comment:
Shouldn't it be a precondition check for logger in the beginnging of method instead of null check here?

Reply:
The idea is that dependencies _may_ not have the logger, and we need not fail if logger is null. So this is typically the pattern we follow in helpers.

Reply:
Should we change it to Option to make our intention explicit?

Reply:
Okay, I will add the null check.. option will make using it a bit awkward